### PR TITLE
[PLATFORM-1342] Use one transaction to update and redeploy publish a product

### PR DESCRIPTION
**Pull request by juhah**
This is ready for testing, just update the docker env to latest version before trying it out. To test:

1) Deploy a paid product, and then publish it normally
2) Unpublish the product
3) Change the price or currency of the product, click Publish

Contract product should updated & published correctly with only one transaction.

### CODE REVIEW

**Comment by juhah** [comment]
PR url with updated staging addresses: http://streamr-marketplace-pr-d94b5cf6bcddaa5a21927f862737d4e564db0fd0.s3-website-eu-west-1.amazonaws.com

File: .travis.yml [hunk 1]
```diff
@@ -153,10 +153,10 @@ jobs:
     - PLATFORM_PUBLIC_PATH=https://marketplace-staging.streamr.com
     - GOOGLE_ANALYTICS_ID=55331565-5
     - STORYBOOK_BASE_PATH=storybook
-    - MARKETPLACE_CONTRACT_ADDRESS=0x7D9d8FE2275a196341DAFf0F277a7B851329590e
+    - MARKETPLACE_CONTRACT_ADDRESS=0x438229f341AEa22f18E69D98c2bE5997bc62d578
```
Comment (juhah) [inline .travis.yml:156]: The new contract (with `redeploy` parameter in `updateProduct`) was not deployed to staging before, this is the correct address and eth watcher has been updated to watch this in staging.

**Comment by fonty1** [comment]
Seems to work, but I think the watcher doesn't pickup the redeploy event properly?
Video:
https://www.dropbox.com/s/e5hdzhtfeeol9ex/Screen%20Recording%202020-05-15%20at%2016.06.38.mov?dl=0

Screenshot 2020-05-15 at 16 16 03

**Comment by fonty1** [comment]
Unrelated but these texts need an update, (redeploy -> republish)
Screenshot 2020-05-15 at 16 00 05
Screenshot 2020-05-15 at 15 59 42
Screenshot 2020-05-15 at 15 59 18

**Comment by juhah** [comment]
> Seems to work, but I think the watcher doesn't pickup the redeploy event properly

I had the same happen locally for the first try, after that it seemed to work. I tried it on staging on that also worked... but could be some flakiness in the ethereum watcher.

**Comment by tumppi** [comment]
Code looks fine but somehow the smart contract calls fail for me. I updated my docker environment but there must be something wrong still 🤔 

I'm getting `MetaMask - RPC Error: Cannot read property 'number' of null` in console.

**Comment by juhah** [comment]
> Unrelated but these texts need an update, (redeploy -> republish)

@fonty1 I've updated the texts for the redeploy case to match publish, republish now only applies when product is already on the marketplace.

> I'm getting MetaMask - RPC Error: Cannot read property 'number' of null in console.

@tumppi I couldn't get this to happen, I wonder if it's just that the account needs resetting? Or Ganache is running as it was replaced with Parity, Docker doesn't automatically stop it. Check also that you have the correct network id in `.env`, marketplace contract address hasn't changed:

```
MARKETPLACE_CONTRACT_ADDRESS=0xF1371c0f40528406dc4f4cAf89924eA9Da49E866
WEB3_REQUIRED_NETWORK_ID=8995
WEB3_PUBLIC_HTTP_PROVIDER=http://localhost:8545
```

With the eth watcher, there seems sometimes to be a timing when starting. At least this way I've been able to get it reliably working:

* Run `streamr-docker-dev stop parity-node0` and `streamr-docker-dev stop ethereum-watcher`
* Run `streamr-docker-dev stop parity-node0` and check from `streamr-docker-dev log -f parity-node0` that it starts mining transactions
* Run  `streamr-docker-dev start ethereum-watcher` and run `streamr-docker-dev log -f ethereum-watcher` until no connection error appears and it starts the watcher:

```
 Connected to Ethereum network:  {"chainId":8995,"name":"unknown"}
 Error: Error: No contract at 0xf1371c0f40528406dc4f4caf89924ea9da49e866
     at throwIfNotContract (/app/src/checkArguments.js:21:15)
     at <anonymous>
     at process._tickDomainCallback (internal/process/next_tick.js:229:7)
 Connected to Ethereum network:  {"chainId":8995,"name":"unknown"}
  Watcher > Checking the Marketplace contract at 0xF1371c0f40528406dc4f4cAf89924eA9Da49E866:
  Watcher >   datacoin: 0xbAA81A0179015bE47Ad439566374F2Bae098686F
  Watcher >   currencyUpdateAgent: 0xa3d1F77ACfF0060F7213D7BF3c7fEC78df847De1
  Watcher >   owner: 0xa3d1F77ACfF0060F7213D7BF3c7fEC78df847De1
  Watcher >   halted: false
  Watcher >   dataPerUsd: 100000000000000000
 Starting watcher...
  Watcher > Starting watcher for Marketplace at 0xF1371c0f40528406dc4f4cAf89924eA9Da49E866```

**Comment by juhah** [comment]
> The only thing I noticed is an edge case bug- if I attempt a republish and something goes wrong with the transaction, the publishing state gets locked on 'publishing'.

@fonty1 Yep, the watcher or the backend should revert back the state if the transaction fails. It's the frontend's job to set that "deploying..." state once we get the transaction hash and eth watcher has to complete the state transfer.

**Comment by juhah** [comment]
Merging this, let's deal with any issues that come up in a separate issue.

**Pull request merged**